pkg/generate/code/set_resource.go

```diff
@@ -1226,53 +1226,61 @@ func SetResourceForStruct(
 	sourceShape := sourceShapeRef.Shape
 	targetShape := targetShapeRef.Shape
 
-	for memberIndex, memberName := range sourceShape.MemberNames() {
-		targetMemberShapeRef := targetShape.MemberRefs[memberName]
-		if targetMemberShapeRef == nil {
+	for _, targetMemberName := range targetShape.MemberNames() {
+		sourceMemberShapeRef := sourceShape.MemberRefs[targetMemberName]
+		if sourceMemberShapeRef == nil {
 			continue
 		}
-		memberVarName := fmt.Sprintf("%sf%d", targetVarName, memberIndex)
-		memberShapeRef := sourceShape.MemberRefs[memberName]
-		memberShape := memberShapeRef.Shape
-		cleanNames := names.New(memberName)
-		sourceAdaptedVarName := sourceVarName + "." + memberName
+		// Upstream logic iterates over sourceShape members and therefore uses
+		// the sourceShape's index; continue using sourceShape's index here for consistency.
+		sourceMemberIndex, err := GetMemberIndex(sourceShape, targetMemberName)
+		if err != nil {
+			continue
```

Comment:
change this to a `panic` so we can get early failure in code generation. I have a feeling it would be super tricky to troubleshoot a problem here if we just ignore and `continue`